docs/iam-usage.rst

```diff
@@ -0,0 +1,177 @@
+- ``iam.ALL_USERS``
+
+.. note::
+  that all of these are convenience wrappers around strings.
+  See the list of member string formats `here <https://cloud.google.com/iam/docs/managing-policies>`_.
+  
+Roles
+~~~~~
+
+Roles represent bundles of permissions that can be added to members.  For a complete list of roles available on a resource run 
+
+``gcloud iam list-grantable-roles //fully/qualified/resource/path``
+
+An ``iam.Role`` object has a name, title, and description
+
+- An ``iam.Role.name`` is the canonical name of a role. This will be the value used as keys in policy dictionaries (see below), and will be referred to as the "role string" throughout this document. E.g. ``'roles/owner'``.
+- ``iam.Role.title`` human readable title of the role. E.g. ``'Owner'``
+- ``iam.Role.description`` the description of a role
+
+Policies
+~~~~~~~~
+
+A policy is a dictionary from role strings to sets of members. ``resource.get_policy()`` also provides etags and versions. An etag is used to provide optimistic concurrency controls on policy updates, while versions are provided for end-user versioning.
+
+
+Policy Changes
+~~~~~~~~~~~~~~
+
+An ``iam.PolicyChange`` object encapsulates changes to made to a policy transactionally. The ``iam.PolicyChange`` constructor takes an optional ``version`` keyword argument, an integer to use as the policy version. If ``version`` is ``None``, when applied ``iam.PolicyChange`` will increment whatever the current version of the policy is by 1.
+
+To apply a ``iam.PolicyChange`` to a resource which implements the IAM interface call the ``apply()`` method with the resource as an argument. 
+
+>>> policy_change = iam.PolicyChange(version=2).add(iam.roles.OWNER.name, [iam.user('alice@example.com')])
+>>> policy, version, etag = policy_change.apply(resource)
+>>> print(policy)
+{'roles/owner': set(['user:alice@example.com'])}
+>>> print(version)
+2
+>>> print(etag)
+xDSFbfdasfAEFdfCds
+
+Apply returns the new policy, it's version, and it's etag.
+
+Optionally, a ``version`` keyword argument can be supplied to ``apply`` which will override the ``version`` behavior of the policy change.
+
+>>> _, version, _ = policy_change.apply(resource, version=None)
+>>> print(version)
+3
+
+Modifications can be added to a ``iam.PolicyChange`` object by one of two methods:
+
+First the user can directly add or remove members from a ``Role``. ``iam.PolicyChange`` exposes two methods for this, ``add`` and ``remove`` which both take a role string or ``iam.Role`` object and a list of member strings.
+
+>>> policy_change = iam.PolicyChange().add(iam.roles.OWNER, [iam.user('alice@example.com')])
+>>> policy_change.remove(iam.roles.EDITOR, [iam.domain('example.com'), iam.group('devs@example.com')])
+>>> policy, _, _ = policy_change.apply(resource)
+>>> print(iam.user('alice@example.com') in policy[iam.roles.OWNER.name])
+True
+>>> print(iam.domain('example.com') in policy[iam.roles.EDITOR.name])
```

Comment:
Since you use `iam.domain('example.com')` (and `iam.user('alice@example.com')`) multiple times in this example, why not assign to them once and then just re-use the variables?